### ISSUE TITLE
Easier way to make data sources available to Public role

### ISSUE THREAD

**Issue by StL-Jim** [dpgaspar/Flask-AppBuilder#435]
It shouldn't be this hard to simply create public views of menu items.  The documentation, while extensive, lacks the ability to _simply_ do this.  

I want a view that anyone can see.  

God if I could just turn security off.  Have no users and just show menu items.  

It shouldn't be this hard!

**Comment by MichaelPereira** [comment]
@centric-jim You should be able to do this by assigning the menu-access permissions of those items to the Public Role that should be automatically created.

**Comment by MichaelPereira** [comment]
Here is how you access it:
<img width="249" alt="screen shot 2017-02-19 at 19 55 00" src="https://cloud.githubusercontent.com/assets/431374/23108584/72710f6e-f6dd-11e6-8d0c-9359ff71a0cd.png">
<img width="1170" alt="screen shot 2017-02-19 at 19 54 51" src="https://cloud.githubusercontent.com/assets/431374/23108585/79b98788-f6dd-11e6-8101-440017b7dd91.png">

**Comment by jkhoogland** [comment]
Hi 
is there any way to explicitly mark a route to be public, 
instead of having to login via the admin account and add the view to the Public role ?

I appreciate the fact that you are forced to explicitly state it is public,
but it is, as mentioned above, quite a hassle for something that should be really straightforward. 
 
Thanks

**Comment by mistercrunch** [comment]
What do you mean by a `route`? Endpoint? Data source?

**Comment by jkhoogland** [comment]
sorry, let me rephrase:

If I declare a view A, 
by default it doesn't show up when running the site,
I need to explicitly add the view A to the Public role in the security settings.

It would be nice if there is some way to declare 
that the view A is automatically added to the Public role.

**Comment by MichaelPereira** [comment]
You can do it in code:

```python
    # Public View Permissions
    views = ['ForgotPasswordResetView']
    for view in views:
        add_view_permissions_to_role(view, role_public)
```

```python
def add_view_permissions_to_role(view, role):
    view_menu = global_app.appbuilder.sm.find_view_menu(view)
    permissions_view_menu = global_app.appbuilder.sm.find_permissions_view_menu(view_menu)
    for permission in permissions_view_menu:
        global_app.appbuilder.sm.add_permission_role(role, permission)
```

**Comment by mistercrunch** [comment]
I'd assume that in most settings you'd want Superset to authenticate users, at least to keep track of  usage. In a POC-type setting I'd assume people can be given credentials to generic accounts.

Anyways, if you bookmark the Public edit role page you're a few clicks away.

Someone could also add a feature similar to `PUBLIC_ROLE_LIKE_GAMMA`, but `PUBLIC_ROLE_LIKE_ALPHA` which would make the Public role the perms equivalent to an Alpha user.

**Comment by mistercrunch** [comment]
BTW this task may belong on the Superset side.
